File: src/main/java/org/cubexmc/metro/train/TrainRuntimeDecisions.java
package org.cubexmc.metro.train;

final class TrainRuntimeDecisions {

    private TrainRuntimeDecisions() {
    }

    static boolean shouldKeepChunksLoaded(boolean chunkLoadingEnabled, boolean globalMode, boolean hasAnyPassengers,
            boolean onlyWhenMoving, TrainInstance.TrainState state) {
        if (!chunkLoadingEnabled) {
            return false;
        }

        if (!globalMode && !hasAnyPassengers) {
            return false;
        }

        if (onlyWhenMoving && state != TrainInstance.TrainState.MOVING) {
            return false;
        }

        return true;
    }
}

File: src/main/java/org/cubexmc/metro/train/TrainRuntimeDecisions.kt
package org.cubexmc.metro.train

internal object TrainRuntimeDecisions {

    @JvmStatic
    fun shouldKeepChunksLoaded(
        chunkLoadingEnabled: Boolean,
        globalMode: Boolean,
        hasAnyPassengers: Boolean,
        onlyWhenMoving: Boolean,
        state: TrainInstance.TrainState?,
    ): Boolean {
        if (!chunkLoadingEnabled) {
            return false
        }

        if (!globalMode && !hasAnyPassengers) {
            return false
        }

        if (onlyWhenMoving && state != TrainInstance.TrainState.MOVING) {
            return false
        }

        return true
    }
}
